### ISSUE TITLE
Generate documentation from FV schema

### ISSUE THREAD

**Issue by erfanium** [icebob/fastest-validator#186]
It would be great to convert FV schema to documentation (e.g: markdown file)
```js
const schema = {
      index: { type: 'number', description: 'Plan order index'  },
      name: 'string',
      price: 'number|min=10',
      durationDays: 'number',
      image: 'string|optional',
      description: 'string|optional',
      priceBeforeDiscount: 'number|optional',
      isFeatured: {
         type: 'boolean',
         default: false,
      },
}

const check = v.compile(schema);
const doc = check.generateSchema();
doc.log(); // using conosle.table
// const md: string = doc.toMarkdown();
``` 
and the output:

Name | type | Description
---- | ----- | --------
`index`  | number | Plan order index
`name` | string | 
`price` | number | min: 10
`durationDays` | number | 
`image` optional | string |  
`description` optional | string | 
`priceBeforeDiscount` optional | number | 
`isFeatured` optional | boolean | default: false

**Comment by icebob** [comment]
Yeah, good idea. But in a separated repo.
I don't understand the `parsedSchema` thing.

**Comment by erfanium** [comment]
@icebob Yes, Most parts will be developed in a separate repo.

**Comment by intech** [comment]
@erfanium hey! Any news?

**Comment by erfanium** [comment]
@intech I did some things, but a little more complicated than I thought (due to the structure of FV).
First we need an API for FV that returns parsed schema (That the document-generator can turn it into a document). this is the hard part!

**Comment by intech** [comment]
@erfanium describe how you see it working, what the inputs and outputs. I will help with the implementation.

**Comment by erfanium** [comment]
for example:
```js
const schema = {
   foo: "string|min:5",
   bar: "number[]",
   o: {
      type: "object",
      props: {
         a: "string|max:24"
      }
   }
}

const unwrapedSchema = FV.unwrapSchema(schema)
```
unwrapedSchema to be:
```js
{
   foo: { type: "string", min: 5 },
   bar: { type: "array", items: "number" },
   o: {
      type: "object",
      props: {
         a: { type: "string", max: 24 }
      }
   }
}
```

**Comment by erfanium** [comment]
@intech
Of course, I really don't know what the proper name for this function is! Do you think `unwrapSchema` or `parseSchema` are good?

**Comment by intech** [comment]
As for the name, I think `parseSchema` or `parse` is logical.

I sketched the code, it still needs to be tested on nested structures.
By the way, in it I used your function, which is not yet in the release.
Am I in the right direction?
https://gist.github.com/intech/eabd6ef917852fea9a2112e675a9068f

**Comment by erfanium** [comment]
@intech Yes, But this `parse` function needs to be in `fv-schema-parser` module, because `FV` itself doesn't use it.

**Comment by erfanium** [comment]
@intech 
Move your codes here: https://github.com/erfanium/fv-document-generator

**Comment by intech** [comment]
@erfanium okey, I'll move this code to the repository and revise it to normal tests
